ecl/hql/hqlfold.cpp

```diff
@@ -5183,8 +5183,12 @@ IHqlExpression * CExprFolderTransformer::doFoldTransformed(IHqlExpression * unfo
     case no_unlikely:
         {
             IHqlExpression * child = expr->queryChild(0);
-            if (child->queryValue())
-                return LINK(child);
+
+            if (child->getOperator()==no_likely || child->getOperator()==no_unlikely)
```

Comment:
A few personal style comments.  Others may not agree
- if the true branch of an if finishes with a return, I don't generally include the else - I find it easier to read 
- an if inside an else clause, I would normally place it in curly brackets {}
- I would probably use a switch statement instead with this new case added.  (child->queryValue() test can be replaced with no_constant).
But none would stop me merging.